### PR TITLE
Create library

### PR DESCRIPTION
**Pull request by lilthree**
Create new library,and bug fix

### CODE REVIEW

File: Pod/Classes/SeafRepos.m [hunk 1]
```diff
@@ -407,4 +407,21 @@ - (BOOL)editable
 {
     return NO;
 }
+
+- (void)createLibrary:(NSString *)newLibName block:(void(^)(bool success, id repoInfo))completeBlock {
+    NSString *requestUrl = [NSString stringWithFormat:API_URL"/repos/"];
+    NSString *form = [NSString stringWithFormat:@"name=%@&desc=%@",newLibName,@"new library"];
```
Comment (poetwang) [inline Pod/Classes/SeafRepos.m]: space between ',' and the following variable.
Certain characters should be replace with percent-escape sequence based on the encoding specified in the newLibName. Use [newLibName escapedUrl]

File: Pod/Classes/SeafRepos.m [hunk 2]
```diff
@@ -407,4 +407,21 @@ - (BOOL)editable
...
+    [connection sendPost:requestUrl form:form
+                 success:
+     ^(NSURLRequest *request, NSHTTPURLResponse *response, id JSON) {
+         Debug("resp=%ld\n", (long)response.statusCode);
+         completeBlock(true,JSON);
```
Comment (poetwang) [inline Pod/Classes/SeafRepos.m]: space
completeBlock(true, JSON);

File: Pod/Classes/SeafRepos.m [hunk 3]
```diff
@@ -407,4 +407,21 @@ - (BOOL)editable
...
+     }
+                 failure:
+     ^(NSURLRequest *request, NSHTTPURLResponse *response, id JSON, NSError *error) {
+         Warning("resp=%ld\n", (long)response.statusCode);
+         [self.delegate download:self failed:error];
```
Comment (poetwang) [inline Pod/Classes/SeafRepos.m]: why  [self.delegate download:self failed:error] only in failure block, not in success block.

File: SeafAction/Info.plist [hunk 4]
```diff
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>XPC!</string>
 	<key>CFBundleShortVersionString</key>
-	<string>2.7.4</string>
+	<string>2.7.5</string>
```
Comment (poetwang) [inline SeafAction/Info.plist]: Next time version bump should be in a separate PR, or directly merged into master.

File: Pod/Classes/SeafRepos.m [hunk 5]
```diff
@@ -407,4 +407,21 @@ - (BOOL)editable
 {
     return NO;
 }
+
+- (void)createLibrary:(NSString *)newLibName block:(void(^)(bool success, id repoInfo))completeBlock {
```
Comment (poetwang) [inline Pod/Classes/SeafRepos.m]: Should the api support encrypted library?

File: seafile/SeafFileViewController.m [hunk 6]
```diff
@@ -313,6 +314,22 @@ - (void)refreshView
         else
             [self noneSelected:NO];
     }
+    if ([_directory isKindOfClass:[SeafRepos class]]) {
```
Comment (poetwang) [inline seafile/SeafFileViewController.m]: Why add this block?
And tempArray could directly fetched from root.items.

File: seafile/SeafFileViewController.m [hunk 7]
```diff
@@ -786,7 +803,11 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 
     if (tableView != self.tableView) {
         // For search results.
-        return [self getSeafFileCell:(SeafFile *)entry forTableView:tableView andIndexPath:indexPath];
+        if ([entry isKindOfClass:[SeafDir class]]) {
```
Comment (poetwang) [inline seafile/SeafFileViewController.m]: For folders in search view, you should use the absolute path as the detail text(the same as file in getSeafFileCell). What happened if user click on the folder?

File: Pod/Classes/SeafRepos.m (hunk 5 above)
Comment (lilthree) [reply Pod/Classes/SeafRepos.m]: api中是否加密是可选项，此处参考了create dir时候的处理，默认就只让创建非加密的了

File: seafile/SeafFileViewController.m (hunk 6 above)
Comment (lilthree) [reply seafile/SeafFileViewController.m]: 当新账号首次登录客户端的没有资料库的时候，root.items为空又回继续调用到showLoadingView，导致loadingview一直在屏幕上不会消失

File: Pod/Classes/SeafRepos.m (hunk 5 above)
Comment (poetwang) [reply Pod/Classes/SeafRepos.m]: dir是否加密取决于它所在的library是否加密，library和dir本来就是不一样的。确定这个我们想要的 iOS上只能创建非加密资料库？Andriod上是怎样的

Comment (freeplant) [reply Pod/Classes/SeafRepos.m]: Android 上是支持创建加密和非加密资料库的。

Comment (lilthree) [reply Pod/Classes/SeafRepos.m]: 我加上创建加密资料库的支持，再上传testflight测试。

**Comment by poetwang** [comment]
一个小问题，iPhone上创建资料库的对话框Encrypted 被截断了 变成 ""Encry..."
